Fix up restoring the selection of revisions on reload

mstv opened the pull request:
Fixes #6057

## Proposed changes

- get the row index from the correct source when it is stable
- clear ToBeSelectedObjectIds again when done

corrects the wrong assumption the revisions were loaded in order
fixes multi-selections not being restored anymore

## Screenshots <!-- Remove this section if PR does not change UI -->

see issue

## Test methodology <!-- How did you ensure quality? -->

- manual

## Test environment(s) <!-- Remove any that don't apply -->

- Git Extensions 3.3.0
- Build 71ee25272ff39bcc59dedca0a4d8fc956cb68971
- Git 2.26.0.windows.1
- Microsoft Windows NT 6.2.9200.0
- .NET Framework 4.8.4180.0
- DPI 96dpi (no scaling)

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

RussKie commented:
@drewnoakes are you still able to repro the issue #6057? Can you try this change and see if it addresses the issue?

drewnoakes commented:
I haven't seen that issue since. If I see it again I'll dig this issue up and comment/reopen.

mstv commented:
> Array.Empty<ObjectId>();

👍 Done.
This PR can be squash-merged.

RussKie commented:
Flip the coin and merge? 🤔

gerhardol commented:
> Flip the coin and merge?

Probably, I have not seen any problems but with limited use. The issue seem to occur like you were stuck in the problematic mode, then released for some reason. (like commit or checkout), so it is unclear if the problem is fixed and that not any new problem occurs.
But probably OK.

mstv commented:
Wrong selections happen quite often with the repos at work.

This PR resolves
- the wrong assumption which was made explicit by #6394 but existed in the code before:
  `ToBeSelectedRowIndexes.Enqueue(_revisionGraph.Count - 1);`
- a more recent regression that multi-selections often are lost on refresh.

The row indexes are retrieved now not before the GUI is told to update.

That's why I recommend the merge into 3.4.

I have squashed and rebased.
